index.html

```diff
@@ -0,0 +1,578 @@
+
+```graphql
+#::[Example of conjunctible forms]
+enum example__Data {}
+
+directive @example(value: String! @deprecated) on SCHEMA
+directive @example(data: example__Data!) on SCHEMA
+```
+
+## Processing Schemas
+
+```mermaid
+#::[Example processing pipeline]
+graph LR
+  schema(["📄  Input Schema"]):::file-->proc("🤖 &nbsp;Processor")
+  proc-->output(["📄  Output Schema"]):::file
+  classDef file fill:none,color:#eee;
+  style proc fill:none,stroke:fuchsia,color:fuchsia;
+```
+
+A common use case is that of a processor which consumes a valid input schema and generates an output schema.
+
+The general guidance for processor behavior is: don't react to what you don't understand.
+
+Specifically, processors:
+  - **should** pass through `@core` directives which reference unknown spec URLs
+  - **should** pass through prefixed directives, types, and other schema elements  
+
+An exception to this is processors which prepare the schema for final public consumption. Such processors **may** choose to eliminate all unknown directives and prefixed types in order to hide schema implementation details within the published schema. This will impair the operation of tooling which relies on these directives—such tools will not be able to run on the output schema, so the benefits and costs of this kind of information hiding should be weighed carefully on a case-by-case basis.
+
+## Validations
+
+### Prefix Uniqueness
+
+**Formal Specification**
+- For each `@core` directive `d` on the SchemaDefinition within the Document,
+  - Let `prefix` be the spec's [prefix](#prefixing) as specified by the directive's `as:` argument or, if the argument is not present, the default prefix specified in the [spec url](#spec-urls).   
+  - For each other `@core` directive `e` where `e ≠ d` on the SchemaDefinition within the Document,
+    - Let `otherPrefix` be the spec's [prefix](#prefixing) as specified by the directive's `as:` argument or, if the argument is not present, the prefix specified in the [spec url](#spec-urls).
+    - `otherPrefix` must not equal `prefix`
+
+**Explanatory Text**
+
+Prefixes, whether implicit or explicit, must be unique within a document. Valid:
+
+<view-of src=prefixing.graphql#schema[0]
+  title="Example of unique prefixes"
+></view-of>
+
+It is also valid to reference multiple versions of the same spec under different prefixes:
+
+<view-of src=prefix-uniqueness.graphql#schema[0]
+  title="Explicit prefixes allow multiple versions of the same spec to coexist within a Document"
+></view-of>
+
+Without the explicit `as:`, the above would be invalid:
+
+<view-of src=prefix-uniqueness.graphql#schema[1]
+  title="Counter-example of non-unique prefixes with multiple versions of the same spec"
+  class=counter-example
+></view-of>
+
+Different specs with the same prefix are also invalid:
```

Comment:
A counter-example worth having (because I'm not sure if you consider it invalid) is two `@core` directives with precisely the same `using` and `as`.

Something that I'm not sure if you want to be valid or not (it's currently valid and I think that's intentional, but worth clarifying) is two `@core` directives with precisely the same `using` and distinct non-null `as` (or one non-null `as` with a value that isn't the default prefix and one null `as`).

Reply:
Yeah, these are good q's that I've run into while writing the core schema libs.

I think we should probably say that two `@core` directives with the same `using` and `as` should probably be invalid, mostly because it does nothing for you. I don't feel super strongly about that though. It's right on the borderline of "maybe this could just be a lint rule?" On the other hand, I think allowing it will (very slightly) increase the complexity of the unique prefix validation, which might be reason enough to just say it's invalid from the get go.

The other case is a bit more interesting: is it ever useful to bring in one spec, at the same version, under two different names? For that matter, is it ever useful to bring in the same major version of a spec twice (say, `join/v1.5` and `join/v1.0`)? I think no, because `join/v1.5` _must_ be backwards compatible with `join/v1.0`, so I could change all of the instances of `join/v1.0` to `join/v1.5` and be exposing exactly the same metadata.

That strongly suggests a simpler validation: every (identity, major version) pair referenced by the document must be uniquely and bidirectionally bound to exactly one prefix. This is rather nice, because it means that I can ask, "what is the prefix of `https://lib.apollo.dev/join/v1.5` within this document?" and get either a single answer, or `null`.